### PR TITLE
feat: reintroduce usePublications

### PR DESCRIPTION
**Pull request by reecejohnson**

### CODE REVIEW

File: examples/web/src/publications/UsePublications.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        {beforeCount > 0 && (
+          <button disabled={loading} onClick={prev}>
+            Fetch newer
+          </button>
+        )}
+        {publications.map((publication) => (
+          <PublicationCard key={publication.id} publication={publication} />
+        ))}
+
+        {hasMore && (
```
Comment (krzysu) [inline examples/web/src/publications/UsePublications.tsx:68]: could also be a good opportunity to check if it works with our `useInfiniteScroll` helper

File: packages/client/src/helpers/buildImageTransformsFromConfig.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { MediaTransformsConfig } from '../consts/config';
+
+export function buildImageTransformsFromConfig(config: MediaTransformsConfig = {}) {
```
Comment (krzysu) [inline packages/client/src/helpers/buildImageTransformsFromConfig.ts:3]: this file is not in use anymore, should be removed

File: packages/client/src/helpers/profileStatistics.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { LensConfig } from '../consts/config';
+import { ProfileStatsArg, ProfileStatsCountOpenActionArgs } from '../graphql/types.generated';
+
+export type ProfileQueryOptions = {
```
Comment (krzysu) [inline packages/client/src/helpers/profileStatistics.ts:4]: I think this file is also not in use anymore, should be removed

File: packages/client/src/types/fetch.d.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+// Simple fetch typings to avoid having to install @types/node-fetch
+// Once @types/node supports fetch api type definitions, we can remove this
```
Comment (krzysu) [inline packages/client/src/types/fetch.d.ts:2]: this file was removed in lens-v2

File: packages/react/src/helpers/arguments.ts [hunk 5]
```diff
@@ -78,3 +81,69 @@ export function useMediaTransformFromConfig<TVariables extends OperationVariable
     ...mediaTransformConfigToQueryVariables(mediaTransforms),
   };
 }
+
+export type WithObserverIdOverride<TVariables = unknown> = Prettify<
```
Comment (krzysu) [inline packages/react/src/helpers/arguments.ts:85]: most of these helpers are not relevant in lens v2 as there is no concept of observer, is any of them in use?

File: packages/react/src/publication/filters.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  restrictPublicationTagsTo?:
+    | {
+        all: string[];
+      }
+    | {
+        oneOf: string[];
+      };
+};
+
+export function createPublicationMetadataFilters(
```
Comment (krzysu) [inline packages/react/src/publication/filters.ts:47]: is this in use?

**Review by krzysu — APPROVED** [review]
cool that you solved the issue!

please remove unnecessary files and helpers that seem like leftovers from v1 or result of merge conflict

File: packages/api-bindings/src/apollo/cache/utils/cursorBasedPagination.ts
Comment (cesarenaldi) [inline packages/api-bindings/src/apollo/cache/utils/cursorBasedPagination.ts:1]: Anything changed in this file? It used to come with data agnostic unit tests... can we have them back in here?

File: examples/web/src/publications/UsePublications.tsx (hunk 1 above)
Comment (reecejohnson) [reply examples/web/src/publications/UsePublications.tsx:68]: good shout, added and works nicely still

File: packages/client/src/types/fetch.d.ts (hunk 4 above)
Comment (reecejohnson) [reply packages/client/src/types/fetch.d.ts]: merge conflicts!

File: packages/api-bindings/src/apollo/cache/utils/cursorBasedPagination.ts
Comment (reecejohnson) [reply packages/api-bindings/src/apollo/cache/utils/cursorBasedPagination.ts:1]: added tests and `moreAfter` back in 👍

File: packages/api-bindings/src/lens/__helpers__/queries.ts [hunk 7]
```diff
@@ -11,9 +12,13 @@ import {
   PublicationVariables,
   PaginatedResultInfo,
 } from '../graphql/generated';
-import { AnyPublication } from '../utils';
+import { AnyPublication, Sources } from '../utils';
 import { mockPaginatedResultInfo, mockPostFragment } from './fragments';
 
+export function mockSources(): Sources {
+  return ['foobar' as AppId];
+}
```
[the remark this answers is not in the document]
Comment (reecejohnson) [reply packages/api-bindings/src/lens/__helpers__/queries.ts]: thanks, removed!

**Pull request merged**